tdh/data_source_backup.go

```diff
@@ -0,0 +1,270 @@
+package tdh
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/svc-bot-mds/terraform-provider-tdh/client/constants/service_type"
+	"github.com/svc-bot-mds/terraform-provider-tdh/client/tdh"
+	"github.com/svc-bot-mds/terraform-provider-tdh/client/tdh/controller"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ datasource.DataSource              = &backupDataSource{}
+	_ datasource.DataSourceWithConfigure = &backupDataSource{}
+)
+
+// NewInstanceTypesDataSource is a helper function to simplify the provider implementation.
+func NewBackupDataSource() datasource.DataSource {
+	return &backupDataSource{}
+}
+
+// instanceTypesDataSource is the data source implementation.
+type backupDataSource struct {
+	client *tdh.Client
+}
+
+// backupResourceModel maps the resource schema data.
+type backupResourceModel struct {
+	ID                types.String   `tfsdk:"id"`
+	Name              types.String   `tfsdk:"name"`
+	ClusterId         types.String   `tfsdk:"cluster_id"`
+	ClusterName       types.String   `tfsdk:"cluster_name"`
+	ClusterVersion    types.String   `tfsdk:"cluster_version"`
+	ServiceType       types.String   `tfsdk:"service_type"`
+	Size              types.String   `tfsdk:"size"`
+	BackupTriggerType types.String   `tfsdk:"backup_trigger_type"`
+	Provider          types.String   `tfsdk:"provider_name"`
+	Region            types.String   `tfsdk:"region"`
+	DataPlaneId       types.String   `tfsdk:"data_plane_id"`
+	Metadata          BackupMetadata `tfsdk:"metadata"`
+	// TODO add upgrade related fields
+}
+
+// clusterMetadataModel maps order item data.
+type BackupMetadata struct {
+	ClusterName    types.String `tfsdk:"cluster_name"`
+	ClusterSize    types.String `tfsdk:"cluster_size"`
+	BackupLocation types.String `tfsdk:"backup_location"`
+}
+
+type backupResponseModel struct {
+	Id          types.String          `tfsdk:"id"`
+	BackupList  []backupResourceModel `tfsdk:"backup_list"`
+	ServiceType types.String          `tfsdk:"service_type"`
+}
+
+type mdsBackupResponse struct {
```

Comment:
we don't need to create this, to get the list of DTOs from paged response, use `(*response).Get()`, refer `resource_object_storage.go`

Reply:
Removed